### PR TITLE
Initial frontend tooling

### PR DESCRIPTION
**Pull request by kaedroho**
Frontend tooling that gives us Typescript, SCSS and React. Taken from wagtail-review.

Planning to build the new edit UI and report in React.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+STATIC_SRC_DIR=wagtail_localize/static_src
+
+npm_deps:
+	cd ${STATIC_SRC_DIR} && npm install
+
+build: npm_deps
+	cd ${STATIC_SRC_DIR} && npm run build
+
+dev: npm_deps
```
Comment (KalobTaulien) [inline Makefile:9]: This will run `npm install` every time we want to watch files, won't it?

File: wagtail_localize/static_src/src/main.tsx [hunk 2]
```diff
@@ -0,0 +1 @@
+// TODO: Add code here
```
Comment (KalobTaulien) [inline wagtail_localize/static_src/src/main.tsx:1]: TODO left behind

File: Makefile (hunk 1 above)
Comment (kaedroho) [reply Makefile:9]: Yep, it finishes really quickly when it hasn't got anything to do so I kinda like running it every time I start the watcher

File: wagtail_localize/static_src/src/main.tsx (hunk 2 above)
Comment (kaedroho) [reply wagtail_localize/static_src/src/main.tsx:1]: Code coming in future PRs!

**Pull request merged**